pylops/medical/ct.py

```diff
@@ -86,54 +94,149 @@ class CT2D(LinearOperator):
     def __init__(
         self,
         dims: InputDimsLike,
-        det_width: int,
-        det_count: float,
+        det_width: float,
+        det_count: int,
         thetas: NDArray,
-        proj_geom_type: Optional[str] = "parallel",
-        source_origin_dist: float = None,
-        origin_detector_dist: float = None,
-        projector_type: Optional[str] = "strip",
-        dtype: DTypeLike = "float64",
+        engine: str,
+        proj_geom_type: str = "parallel",
+        source_origin_dist: Optional[float] = None,
+        origin_detector_dist: Optional[float] = None,
+        projector_type: Optional[str] = None,
+        dtype: DTypeLike = "float32",
         name: str = "C",
     ) -> None:
         if astra_message is not None:
             raise NotImplementedError(astra_message)
 
-        # create volume and projection geometries
-        self.vol_geom = astra.create_vol_geom(dims)
-        if proj_geom_type == "parallel":
+        self.dims = dims
+        self.det_width = det_width
+        self.det_count = det_count
+        self.thetas = to_numpy(thetas)  # ASTRA can only consume angles as a NumPy array
+        self.engine = engine
+        self.proj_geom_type = proj_geom_type
+        self.source_origin_dist = source_origin_dist
+        self.origin_detector_dist = origin_detector_dist
+
+        # make "strip" projector type default for cpu engine and only allow cuda otherwise
+        if engine == "cpu":
+            if projector_type is None:
+                projector_type = "strip"
+            # fanflat geometry projectors need an appropriate suffix (unless it's "cuda")
+            if projector_type == "cuda":
+                logging.warning("'cuda' projector type specified with 'cpu' engine.")
```

Comment:
Please use this logic (https://github.com/PyLops/pylops/blob/18321da8f07d0d4fc897400cd1867f56f70001f9/pylops/basicoperators/sum.py#L13) to be consistent with the rest of PyLops' logging messages